src/words/environment/WordsObject.java

```diff
@@ -84,8 +108,16 @@ public void setProperty(String propertyName, Property property) throws WordsRunt
 
 		if (properties.containsKey(propertyName) && property.type == Property.PropertyType.NOTHING)
 			properties.remove(propertyName);
```

Comment:
This is actually what I consider the slickest part of my implementation. Your concern is valid, in addition to the concern if object A's property X refers object B, and then we set object A's property X to something else, shouldn't object A be removed as a referer to object B?

This is where our property type of "NOTHING" comes in to save the day. Since we decided not having a property assigned is the same has having NOTHING assigned, the clearReferers() method in the WordsObject class simply goes through all the Property objects and converts them to NOTHING. So if object A's X was pointing at object B, and we remove B, A's X is now pointing at NOTHING. If A's X was pointing at B, and then A's X was removed or set to some other property, it is no longer pointing at object B. Nevertheless, the original property will still be converted to NOTHING in clearReferers().